whole-tale/ngx-dashboard
# Fix: add/style missing info on metadata

## Problem
Metadata view is missing some necessary information and styling does not very closely match the mockups.

Fixes #47 
Fixes #17 (but may still need very minor spacing fixes on read-only metadata view)

## Approach
Replace edit/save/cancel with static views that more closely match the mockups provided [here](https://whole-tale.github.io/wholetale-css-mockup/src/run-meta.html).

## How to Test
Prerequisites: at least one Tale created (Owned Tale, does not need to be running), at least one dataset registered

1. Checkout this branch locally, rebuild the dashboard
2. Login to the WholeTale Dashboard
3. Navigate to the Tale Catalog
4. Locate a Tale that you own and click View
5. Click the Files tab to go to the Run > Files view, then attach one or more External Datasets to the Tale
6. Click the Metadata tab to go to the Run > Metadata view
    * You should see a read/write view of the Tale's metadata, allowing you to edit the fields and Save
    * You should see that required `input` fields now show a green border on their left side
    * You should see Citations listed for each of the datasets you've added to the Tale
7. Add an author without filling out Name or ORCID, then scroll to the bottom
    * You should see that invalid fields, such as an `input` that is `required` but empty, are given a red border
    * You should see that the Save button is disabled if a required field is empty
8. Fill in the author's first name, last name, and ORCID
9. Modify the title, category, environment, license, description, and illustration (optional) and click Save
    * You should **not** be able to clear the value of the environment or license dropdowns
    * You should see a notification indicating that the Tale was saved successfully
